ozaria/site/views/play/level/tome/SpellPaletteView.coffee

```diff
@@ -28,114 +23,39 @@ module.exports = class SpellPaletteView extends CocoView
     @aceEditors = []
-    docs = @options.level.get('documentation') ? {}
-    @showsHelp = docs.specificArticles?.length or docs.generalArticles?.length
     @createPalette()
     $(window).on 'resize', @onResize
 
   getRenderData: ->
     c = super()
     c.entryGroups = @entryGroups
-    c.entryGroupSlugs = @entryGroupSlugs
-    c.entryGroupNames = @entryGroupNames
-    c.tabbed = _.size(@entryGroups) > 1
-    c.defaultGroupSlug = @defaultGroupSlug
-    c.showsHelp = @showsHelp
-    c.tabs = @tabs  # For hero-based, non-this-owned tabs like Vector, Math, etc.
-    c.thisName = {coffeescript: '@', lua: 'self', python: 'self', java: 'hero'}[@options.language] or 'this'
     c._ = _
     c
 
   afterRender: ->
     super()
-    if @entryGroupSlugs
-      for group, entries of @entryGroups
-        groupSlug = @entryGroupSlugs[group]
-        for columnNumber, entryColumn of entries
-          col = $('<div class="property-entry-column"></div>').appendTo @$el.find(".properties-#{groupSlug}")
-          for entry in entryColumn
-            col.append entry.el
-            entry.render()  # Render after appending so that we can access parent container for popover
-      @$('.nano').nanoScroller alwaysVisible: true
-      @updateCodeLanguage @options.language
-    else
-      @entryGroupElements = {}
-      for group, entries of @entryGroups
-        @entryGroupElements[group] = itemGroup = $('<div class="property-entry-item-group"></div>').appendTo @$el.find('.properties-this')
-        if entries[0].options.item?.getPortraitURL
-          itemImage = $('<img class="item-image" draggable=false></img>').attr('src', entries[0].options.item.getPortraitURL())
-          itemGroup.append itemImage
-          firstEntry = entries[0]
-          do (firstEntry) ->
-            itemImage.on "mouseenter", (e) -> firstEntry.onMouseEnter e
-            itemImage.on "mouseleave", (e) -> firstEntry.onMouseLeave e
-        for entry, entryIndex in entries
-          itemGroup.append entry.el
-          entry.render()  # Render after appending so that we can access parent container for popover
-          if entries.length is 1
-            entry.$el.addClass 'single-entry'
-          if entryIndex is 0
-            entry.$el.addClass 'first-entry'
-      for tab, entries of @tabs or {}
-        tabSlug = _.string.slugify tab
-        itemsInGroup = 0
+    for group, entries of @entryGroups
+      group = _.string.slugify(group)
+      itemGroup = $('<div class="property-entry-item-group"></div>').appendTo @$el.find('.properties-'+group)
+      entrySubGroups = _.groupBy entries, (entry) -> entry.doc.subSection || 'none'
+      for subGroup, entries of entrySubGroups
+        if subGroup != 'none'
+          header = $("<a class='sub-section-header btn btn-small btn-illustrated' data-panel='#sub-section-#{subGroup}-#{group}'><span>#{subGroup}</span><div style='float: right' class='glyphicon glyphicon-chevron-down'></div></a>").appendTo itemGroup
```

Comment:
Could this be split across multiple lines?

Reply:
yeah thats a good idea! 👍